app/src/js/actions/index.js

```diff
@@ -383,6 +383,41 @@ export const deleteGranuleClearError = (granuleId) => ({
   id: granuleId
 });
 
+export const removeAndDeleteGranule = (granuleId) => (dispatch, getState) => {
+  const granules = getState().granules.list.data;
+  const granuleToDelete = granules.find((g) => g.granuleId === granuleId);
+  const errorPresent = (response, errorType) => response && response.type && response.type === errorType;
+
+  // If the granule is published to CMR, remove it from CMR and then delete.
+  if (granuleToDelete.published) {
+    return dispatch(removeGranule(granuleId))
+      .then(
+        (removeResponse) => {
+          if (errorPresent(removeResponse, types.GRANULE_REMOVE_ERROR)) {
+            // If the remove request is unsuccessful, we still dispatch a DELETE error because
+            // that is what the action is expecting.
+            return dispatch({
+              type: types.GRANULE_DELETE_ERROR,
+              id: granuleId,
+              error: removeResponse.error
+            });
+          }
+        }
+      )
+      .then(
+        (removeResponse) => {
+          // If the previous remove request was successful, delete the granule
+          if (!errorPresent(removeResponse, types.GRANULE_DELETE_ERROR)) {
+            return dispatch(deleteGranule(granuleId));
+          }
+        }
+      );
+  }
+
+  // If this granule is NOT published, just delete it
+  return dispatch(deleteGranule(granuleId));
+};
```

Comment:
Totally open to handling this series of synchronous calls differently. I'm replicating the approach I would take using async/await but there may be a better Redux/Thunk approach that I'm not seeing.

Reply:
@dopeters What do you think of this approach?

Reply:
I think this approach makes sense and follows the pattern we are using elsewhere. No issues from my perspective!